app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/interactor/SwapTokensInteractor.kt

```diff
@@ -39,6 +55,37 @@ class SwapTokensInteractor(
         return allTokens.withoutCurrentToken(tokenA.mintAddress)
     }
 
+    suspend fun getAllAvailableTokensB(): List<SwapTokenModel> {
+        val userTokens = homeLocalRepository.getUserTokens().map(SwapTokenModel::UserToken)
+        val jupiterTokens = swapTokensRepository.getTokens().map(SwapTokenModel::JupiterToken)
+
+        val tokenA = getCurrentTokenA()
+        val tokenB = getCurrentTokenB()
+        val availableTokenBMints = swapRoutesRepository.getSwappableTokenMints(sourceTokenMint = tokenA.mintAddress)
+
+        val allTokensB = userTokens + jupiterTokens
+        return allTokensB
+            .withoutCurrentToken(tokenB.mintAddress)
+            .filter { it.mintAddress in availableTokenBMints }
```

Comment:
Would be better to filter only once, for example: removing current token in `filter`
You're creating a new list in `withoutCurrentToken` and in `filter` here and everywhere you are using it

Reply:
okay